### PR TITLE
Add markdown linting action

### PR DESCRIPTION
**Pull request by thatblindgeye**
## Because

Making updates to lesson layouts/styles will only go so far if those updates are undone. Adding an action to check for a lot of the styling/layout for lessons will help prevent regressions/future issues being merged in.

## This PR

- Adds a GH action to lint markdown files, calling out errors in any files updates

## Issue

Closes #XXXXX

## Pull Request Requirements

-   [ ] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
-   [ ] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`
-   [ ] The `Because` section summarizes the reason for this PR
-   [ ] The `This PR` section has a bullet point list describing the changes in this PR
-   [ ] If this PR addresses an open issue, it is linked in the `Issue` section
-   [ ] If any lesson files are included in this PR, they have been previewed with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure it is formatted correctly
-   [ ] If any lesson files are included in this PR, they follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

### CODE REVIEW

**Review by ManonLef — COMMENTED** [review]
@thatblindgeye nice work! Do you think we should include newlines before and after headings like `MD022` ?

I think we could possibly add `MD009 no trailing spaces` as well since this is something I do come across quite a bit.

This would be a nice addition. I am not entirely sure how to check the actual workings on this in the most straightforward way. Perhaps we can add it to a team repo first to test or do you think we can test in prod without breaking things :D?

**Comment by ManonLef** [comment]
Oh and I noticed the linter throwing a typo on "errors" in line 17 which I forgot to include in my comment.

**Comment by thatblindgeye** [comment]
@ManonLef Any default configs should still apply. The jsonc file just uses custom settings for things.

We could use https://github.com/thatblindgeye/curriculum/pull/7 as a basis. Just a PR against my own repo with similar config and workflow files. I just made another change to that PR to remove the spacing before a heading, and the workflow catches it as an error in the files.

It shouldn't break anything since I'm not enabling automatic fixes in the workflow. If we were to merge it now the worst case is most PRs will have errors flagged, which we could just ignore for now. We could also just wait to merge this until most updates are made manually, then just have this ensure we didn't miss anything/nothing else sneaks in.

**Comment by ManonLef** [comment]
Oh I'm sorry @thatblindgeye for not fully understanding perhaps. I have never before looked into GitHub linting PR's. I see on your linked repo it also does throw a check error on MD022 for example. I now realise that as I understand it, these are the customisations you drafted up and the other MD### items are also included in their default status?

**Comment by thatblindgeye** [comment]
@ManonLef yep! I wasn't sure until I tested though so thank you for mentioning that 😆

**Comment by ManonLef** [comment]
It looks pretty useful since it also shows the lines on which the error was thrown which makes it easy to propose a change to that line which can be directly committed. 

Do you know why it is showing all errors twice?

Screenshot 2023-11-04 at 17 28 38

**Comment by ManonLef** [comment]
I'm assuming it's because you're testing two similar linters, but checking to be sure :)

Screenshot 2023-11-04 at 17 31 02

**Comment by thatblindgeye** [comment]
Yeah I believe that's why, the file in this PR only runs on `pull_request` so that should prevent those duplicate errors. That was just an error in my tinkering with things 😆

**Review by ManonLef — APPROVED** [review]

**Review by wise-king-sullyman — APPROVED** [review]

File: .github/workflows/markdownlint.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+name: MarkdownLint
+on: pull_request
+
+jobs:
+  lint:
```
Comment (thatblindgeye) [inline .github/workflows/markdownlint.yml:5]: Copy paste this job and have 2 jobs: one for linting lesson files and another for project files

**Review by wise-king-sullyman — APPROVED** [review]

**Pull request merged**